src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -666,7 +694,30 @@ public void getSigInsertionIndex_whenMalformedRedeemScript_shouldReturnZero() {
         BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
 
         int sigInsertionIndex = customRedeemScript.getSigInsertionIndex(hashForSignature, signingKey);
-        Assert.assertEquals(EXPECTED_DEFAULT_SIG_INSERTION_INDEX_FOR_NO_REDEEM_SCRIPT, sigInsertionIndex);
+        // This return zero because the NoRedeemScriptParser.findKeyInRedeem is being used and is
+        // returning -1 as default value instead of throwing an exception
```

Comment:
Is this right?

Reply:
This is just the current implementation that we have for the NonStandardRedeemScriptParser. But it isn't right if we compare it with the [bitcoinj implementation](https://github.com/rsksmart/bitcoinj/blob/3e0a65f36f10062b8b3247204715b6b370158666/core/src/main/java/org/bitcoinj/script/Script.java#L474) which throws an `IllegalStateException` if the key is not found.